tests/jax/components/training/losses_test.py

```diff
@@ -0,0 +1,191 @@
+
+    return test_mapg
+
+
+def test_mapg_creation(
+    mock_trainer: Trainer,
+    mapg_trust_region_clipping_loss: MAPGWithTrustRegionClippingLoss,
+) -> None:
+    """Test whether mapg function is successfully created"""
+
+    mapg_trust_region_clipping_loss.on_training_loss_fns(trainer=mock_trainer)
+    assert hasattr(mock_trainer.store, "grad_fn")
+    assert isinstance(mock_trainer.store.grad_fn, Callable)  # type:ignore
+
+
+def test_mapg_loss(
+    mock_trainer: Trainer,
+    mapg_trust_region_clipping_loss: MAPGWithTrustRegionClippingLoss,
+) -> None:
+    """Test whether mapg loss output is as expected"""
+    mapg_trust_region_clipping_loss.on_training_loss_fns(trainer=mock_trainer)
+    grad_fn = mock_trainer.store.grad_fn
+
+    actions = {
+        agent: jnp.array([1.0, 1.0, 1.0, 1.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    behaviour_log_probs = {
+        agent: jnp.array([-2.0, -2.0, -2.0, -2.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    target_values = {
+        agent: jnp.array([3.0, 3.0, 3.0, 3.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    advantages = {
+        agent: jnp.array([2.0, 2.0, 2.0, 2.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    behavior_values = {
+        agent: jnp.array([1.0, 1.0, 1.0, 1.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+
+    low_target_values = {
+        agent: jnp.array([1.0, 1.0, 1.0, 1.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    low_advantages = {
+        agent: jnp.array([3.0, 3.0, 3.0, 3.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+
+    _, loss_info = grad_fn(
+        params=mock_trainer.store.parameters,
+        observations=mock_trainer.store.observations,
+        actions=actions,
+        behaviour_log_probs=behaviour_log_probs,
+        target_values=target_values,
+        advantages=advantages,
+        behavior_values=behavior_values,
+    )
+
+    _, low_loss_info = grad_fn(
+        params=mock_trainer.store.parameters,
+        observations=mock_trainer.store.observations,
+        actions=actions,
+        behaviour_log_probs=behaviour_log_probs,
+        target_values=low_target_values,
+        advantages=low_advantages,
+        behavior_values=behavior_values,
+    )
+
+    agent_0_loss = loss_info["agent_0"]
+    loss_entropy = agent_0_loss["loss_entropy"]
+    loss_value = agent_0_loss["loss_value"]
+    loss_policy = agent_0_loss["loss_policy"]
+    loss_total = agent_0_loss["loss_total"]
+
+    low_agent_0_loss = low_loss_info["agent_0"]
+    low_loss_value = low_agent_0_loss["loss_value"]
+    low_loss_policy = low_agent_0_loss["loss_policy"]
+
+    assert loss_entropy == -0.47500002
+    assert loss_value == 6.4075003
+    assert loss_total == (loss_entropy * 0.01 + loss_policy + loss_value * 0.5)
+
+    assert low_loss_value < loss_value
+    assert low_loss_policy < loss_policy
```

Comment:
This is still confusing me a little bit, since we're asserting for a value of `loss_total` here?

Reply:
I was checking if the `loss_total` was being added together correctly based on the default coefficients in the config. I've removed it if you think it's necessary?

Reply:
I've changed it to check if the `low_loss_total' is less than the `loss_total` value.

Reply:
It gets the `policy_loss` value from rlax but it's just a summation of the other 3 losses. So if the other 3 losses are working as intended then the check is just too see that it's being generated as intended.

Reply:
@siddarthsingh1 this still requires attention. You can add back in the direct assertion for loss values.

Reply:
Added it back

Reply:
Awesome, if we're doing this then, do we even still need the `low_loss_total` etc. assertions?

Reply:
@siddarthsingh1 what are your thoughts? Seems to me that if we can directly assert for values like above, then we don't need this high / low loss value setup

Reply:
Thanks for clarifying this morning. I'm happy with how it is now, makes sense!